cpuid/src/common.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::arch::x86_64::CpuidResult;
+use std::arch::x86_64::__cpuid_count;
+use std::arch::x86_64::__get_cpuid_max;
+
+use kvm::CpuId;
+use kvm_bindings::kvm_cpuid_entry2;
+
+const INTEL: &[u8; 12] = b"GenuineIntel";
+const AMD: &[u8; 12] = b"AuthenticAMD";
+
+const EXT_FUNCTION: u32 = 0x80000000;
+
+pub enum Error {
+    InvalidParameters(String),
+    NotSupported,
+    SizeLimitExceeded,
+}
+
+fn get_cpuid(function: u32, count: u32) -> Result<CpuidResult, Error> {
+    if unsafe { __get_cpuid_max(function & EXT_FUNCTION).0 } < function {
```

Comment:
Each unsafe block should be precede by an explanation why is it safe to use.

Reply:
Good point. It seems I was also missing some cases here. Fixed